projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.service.ts

```diff
@@ -51,21 +75,64 @@ export class MerchandisingCarouselComponentService {
     );
   }
 
+  sendCarouselViewEvent(
+    merchandisingCarouselModel: MerchandisingCarouselModel
+  ): void {
+    const carouselEvent: CarouselEvent = this.getCarouselEventFromCarouselModel(
+      merchandisingCarouselModel
+    );
+    const productSkus: string[] = [];
+    merchandisingCarouselModel.items$.forEach((merchandisingProduct$) =>
+      merchandisingProduct$
+        .pipe(
+          filter((merchandisingProduct) => Boolean(merchandisingProduct.code))
+        )
+        .subscribe((merchandisingProduct) =>
+          productSkus.push(merchandisingProduct.code)
+        )
+    );
+
+    this.profileTagEventService.notifyProfileTagOfEventOccurence(
+      new MerchandisingCarouselViewed(carouselEvent, productSkus)
+    );
+  }
+
+  sendCarouselItemClickedEvent(
+    merchandisingCarouselModel: MerchandisingCarouselModel,
+    clickedProduct: MerchandisingProduct
+  ): void {
+    const carouselEvent: CarouselEvent = this.getCarouselEventFromCarouselModel(
+      merchandisingCarouselModel
+    );
+
+    carouselEvent.metadata = {
+      ...carouselEvent.metadata,
+      ...clickedProduct.metadata,
+    };
+
+    this.profileTagEventService.notifyProfileTagOfEventOccurence(
```

Comment:
I was just coming to have a look to see if you were calling the event layer directly, or using our event service. Great job :+1: